File: ui/src/app/shared/board.ts
export class Board {
  boardCellStatuses: Array<number[]>;
  rows: number;
  columns: number;
  occupationCount:number;
  hitCount: number;
  hasLost: boolean;
}

export class AttackInfo {
  board: Board;
  row: number;
  column: number;
}

export enum AttackResult {
  hit,
  miss
}
 export enum BoardCellStatus {
   hit,
   miss,
   unoccupied,
   occupied
 }

 export enum GameMode {
   on,
   quit,
   finished
 }

File: Entities/Board.cs
using System.Linq;
using System.Linq.Dynamic.Core;
using BattleshipTest.Enums;
using Microsoft.EntityFrameworkCore.Internal;

namespace BattleshipTest.Entities
{
    public class Board
    {
        public BoardCellStatus[,] BoardCellStatuses { get; set; }
        public int Rows { get; set; }
        public int Columns { get; set; }
        public int OccupationCount { get; set; }
        public int HitCount { get; set; }
    }
}